MaterialLib/SolidModels/ViscoElasticBurgers-impl.h

```diff
@@ -0,0 +1,332 @@
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PHYSICALQUANITIES_SOLIDS_VISCOELASTICBURGERS_IMPL_H_
+#define PHYSICALQUANITIES_SOLIDS_VISCOELASTICBURGERS_IMPL_H_
+
+#include <logog/include/logog.hpp>
+
+#include "MaterialLib/SolidModels/KelvinVector.h"
+
+namespace Solids
+{
+template <int DisplacementDim>
+void ViscoElasticBurgers<DisplacementDim>::computeConstitutiveRelation(
+    double const t,
+    ProcessLib::SpatialPosition const& x,
+    double const dt,
+    typename ViscoElasticBurgers<
+        DisplacementDim>::KelvinVector const& /*eps_prev*/,
+    typename ViscoElasticBurgers<DisplacementDim>::KelvinVector const& eps,
+    typename ViscoElasticBurgers<
+        DisplacementDim>::KelvinVector const& /*sigma_prev*/,
+    typename ViscoElasticBurgers<DisplacementDim>::KelvinVector& sigma,
+    typename ViscoElasticBurgers<DisplacementDim>::KelvinMatrix& C,
+    typename MechanicsBase<DisplacementDim>::MaterialStateVariables&
+        material_state_variables)
+{
+    assert(dynamic_cast<MaterialStateVariables*>(&material_state_variables) !=
+           nullptr);
+    MaterialStateVariables& _state =
+        static_cast<MaterialStateVariables&>(material_state_variables);
+    // calculation of deviatoric parts
+    auto const& P_dev = MaterialLib::SolidModels::Invariants<
+        KelvinVectorSize>::deviatoric_projection;
+    KelvinVector const epsd_i = P_dev * eps;
+
+    // initial guess as elastic predictor
+    KelvinVector sigd_j = 2.0 * (epsd_i - _state._eps_M_t - _state._eps_K_t);
+
+    // Calculate effective stress and update material properties
+    double sig_eff = MaterialLib::SolidModels::Invariants<
+        KelvinVectorSize>::equivalentStress(sigd_j);
+    UpdateBurgersProperties(t, x, sig_eff * _state.GM, _state);
+
+    // initial evaluation of residual and Jacobian
+    Eigen::Matrix<double, KelvinVectorSize * 3, 1> res_loc;
+    Eigen::Matrix<double, KelvinVectorSize * 3, KelvinVectorSize * 3,
+                  Eigen::RowMajor>
+        K_loc;
+    CalResidualBurgers(dt, epsd_i, sigd_j, _state._eps_K_j, _state._eps_K_t,
+                       _state._eps_M_j, _state._eps_M_t, res_loc, _state);
+    CalJacobianBurgers(t, x, dt, K_loc, sig_eff, sigd_j, _state._eps_K_j,
+                       _state);
+
+    //
+    // Local Newton loop
+    //
+
+    int counter = 0;
+    const int counter_max(20);
+    const double local_tolerance(1.e-20);
+
+    // Linear solver for the newton loop is required after the loop with the
+    // same matrix. This saves one decomposition.
+    Eigen::PartialPivLU<Eigen::Matrix<double, KelvinVectorSize * 3,
+                                      KelvinVectorSize * 3, Eigen::RowMajor>>
+        linear_solver(KelvinVectorSize * 3);
+
+    while (res_loc.squaredNorm() > local_tolerance && counter < counter_max)
```

Comment:
suggest using a common Newton Raphson sovler instead of implementing this for each case